src/sagemaker/jumpstart/accessors.py

```diff
@@ -90,11 +101,20 @@ def _get_manifest(region: str = JUMPSTART_DEFAULT_REGION_NAME) -> List[JumpStart
         Args:
             region (str): Optional. The region to use for the cache.
         """
+
+        old_cache_kwargs = JumpStartModelsAccessor._cache_kwargs.copy()
+
+        additional_kwargs = {}
+        if s3_client is not None:
+            additional_kwargs.update({"s3_client": s3_client})
+
         cache_kwargs = JumpStartModelsAccessor._validate_and_mutate_region_cache_kwargs(
-            JumpStartModelsAccessor._cache_kwargs, region
+            {**JumpStartModelsAccessor._cache_kwargs, **additional_kwargs}, region
         )
         JumpStartModelsAccessor._set_cache_and_region(region, cache_kwargs)
-        return JumpStartModelsAccessor._cache.get_manifest()  # type: ignore
+        manifest = JumpStartModelsAccessor._cache.get_manifest()  # type: ignore
+        JumpStartModelsAccessor._cache_kwargs = old_cache_kwargs
```

Comment:
Quick question - why are we setting the `cache_kwargs` value to the old value? Correct me if I'm wrong but would this make the `or new_cache_kwargs != JumpStartModelsAccessor._cache_kwargs` check above fail every time, resetting the cache each time?

Reply:
These methods are not intended to mutate the cache kwargs itself, there's another method for that `set_cache_kwargs`.